var/spack/repos/builtin/packages/macsio/package.py

```diff
@@ -0,0 +1,78 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Macsio(CMakePackage):
+    """A Multi-purpose, Application-Centric, Scalable I/O Proxy Application"""
+
+    homepage = "http://llnl.github.io/MACSio"
+
+    # NOTE: macsio-v0.9 is build with gmake and is not properly built here.
+    # url = "https://github.com/LLNL/MACSio/releases/download/v0.9/macsio-0.9.tar.gz"
+    # version('0.9', '6d4bf863c90975a3df54795c5abb9eb8f23d5aaf')
+
+    version('master', git='https://github.com/LLNL/MACSio.git', branch='cmake2')
+    depends_on('json-cwx')
+
+    ## TODO: Test these and add these to spack!
+
+    variant('mpi', default=False, description="Build MPI plugin")
+    depends_on('mpi', when="+mpi")
```

Comment:
in most other packages the code first lists all variants and then all `depends_on`.